### ISSUE TITLE
Faraday::SSLError: hostname does not match the server certificate with warn_level: :store

### ISSUE THREAD

**Issue by dentarg** [jaimeiniesta/metainspector#149]
With `warn_level: :store`, I don't think this should happen, right?

```ruby
[2] pry(main)> MetaInspector.new("https://www.blogblogyaquelquun.com", { warn_level: :store }).ok?
Faraday::SSLError: hostname "www.blogblogyaquelquun.com" does not match the server certificate
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/openssl/ssl.rb:240:in `post_connection_check'
```

Additional info:

```ruby
[1] pry(main)> MetaInspector.new("https://www.blogblogyaquelquun.com", { warn_level: :store, html_content_only: true }).ok?
Faraday::SSLError: hostname "www.blogblogyaquelquun.com" does not match the server certificate
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/openssl/ssl.rb:240:in `post_connection_check'
```

```ruby
[3] pry(main)> MetaInspector.new("https://www.blogblogyaquelquun.com", { }).ok?
Faraday::SSLError: hostname "www.blogblogyaquelquun.com" does not match the server certificate
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/openssl/ssl.rb:240:in `post_connection_check'
```

**Comment by jaimeiniesta** [comment]
You found a similar issue there! I'll take a look as soon as I can.

**Comment by dentarg** [comment]
Here's another one, same type of exception, but different cause: "certificate verify failed"

```ruby
[1] pry(main)> MetaInspector.new("https://www.follidee.com", { }).ok?
Faraday::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/net/http.rb:923:in `connect'
```

```ruby
[2] pry(main)> MetaInspector.new("https://www.follidee.com", { warn_level: :store }).ok?
Faraday::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/net/http.rb:923:in `connect'
```

```ruby
[3] pry(main)> MetaInspector.new("https://www.follidee.com", { warn_level: :store, html_content_only: true }).ok?
Faraday::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
from /Users/dentarg/.rubies/ruby-2.2.3/lib/ruby/2.2.0/net/http.rb:923:in `connect'
[4] pry(main)>
```

**Comment by jaimeiniesta** [comment]
Thank you! We were not considering the `Faraday::SSLError` exception.

Fixed and released on 4.7.2

**Comment by dentarg** [comment]
Thanks :) :+1:
